util.py

```diff
@@ -86,6 +86,34 @@ def status_to_string(status: int) -> set:
 	except IndexError:
 		raise ValueError("invalid value for status")
 
+def type_to_string(elemental_type: int) -> set[str]:
+	"""Converts a bit mask of elemental types to human readable strings.
+
+	:param elemental_type: A bit mask of elemental types.
+	:returns: A set of all elemental types indicated by the bit mask.
+	"""
+
+	elements = [
+		"Normal",
+		"Fighting",
+		"Flying",
+		"Poison",
+		"Ground",
+		"Rock",
+		"Bug",
+		"Ghost",
+		"Steel",
+		"Fire",
+		"Water",
+		"Grass",
+		"Electric",
+		"Psychic",
+		"Ice",
+		"Dragon",
+		"Dark",
+	]
+	return set([flag for (index, flag) in enumerate(elements) if (elemental_type & 1 << index)])
```

Comment:
This all looks good to me, but should this have error handling similar to `status_to_string`?

Reply:
In this case, it's not possible to get an index out of bounds because we are iterating over `elements`, so we are inherently restricted by the bounds of `elements`. If you give this function `1 << 22`, it will simply give you an empty set.

Reply:
👍